src/ert/config/observations.py

```diff
@@ -234,13 +228,23 @@ def _get_restart(
                 f"Missing REFCASE or TIME_MAP for observations: {obs_name}",
                 obs_name,
             )
-        time, date_str = EnkfObs._get_time(date_dict, time_map[0])
+
+        try:
+            time, date_str = EnkfObs._get_time(date_dict, time_map[0])
+        except ObservationConfigError:
+            raise
+        except ValueError as err:
+            raise ObservationConfigError.with_context(
+                f"Failed to parse date of {obs_name}", obs_name
+            ) from err
+
         try:
             return EnkfObs._find_nearest(time_map, time)
         except IndexError as err:
-            raise IndexError(
+            raise ObservationConfigError.with_context(
                 f"Could not find {time} ({date_str}) in "
-                f"the time map for observation {obs_name}"
+                f"the time map for observations {obs_name}",
```

Comment:
When you remove the except IndexError in `ert_config.py`, this is where you would add back the message.

```suggestion
                f"the time map for observations {obs_name}",
                 "The time map is set from the REFCASE keyword. Either "
                 "the REFCASE has an incorrect/missing date, or the observation "
                  "is given an incorrect date."
```

Reply:
added suggestion with some minor changes